### ISSUE TITLE
[Bug]: OneSignal.OSOperationRepo EXC_BAD_ACCESS KERN_INVALID_ADDRESS 0x0000000000000010

### ISSUE THREAD

**Issue by sebastinto** [OneSignal/OneSignal-iOS-SDK#1409]
### What happened?

Got notified of this isolated crash by Crashlytics (see attached log output).

Since this happened on v.5.1.3 of the OneSignal SDK, we will update to 5.1.5 but still wanted to report the crash in case it's helpful. 

If there's anything we can do to either address or help in investigating this crash, please don't hesitate to let us know, thanks!

### Steps to reproduce?

```Markdown
Not reproducible atm.
```

### What did you expect to happen?

No crash. :)

### OneSignal iOS SDK version

5.1.3

### iOS version

15

### Specific iOS version

```Markdown
iPadOS 16.6.0
```

### Relevant log output

```Shell
# Crashlytics - Stack trace
# Application: com.myapp.myapp
# Platform: apple
# Version: {redacted}
# Issue: b080f49da7bb072c4a0cbf850b2189ee
# Session: 875bd90584a94207830af8a989ec4aea_DNE_0_v2
# Date: Mon Apr 15 2024 14:36:31 GMT-0600 (Mountain Daylight Time)

com.apple.main-thread
0  SwiftUI                        0x4162c __swift_memcpy8_8 + 918
1  SwiftUI                        0x4958 -[NSAttributedString(SPI) initSwiftUIAttributedStringWithFormat:options:locale:arguments:] + 132
2  SwiftUI                        0xd69c00 OUTLINED_FUNCTION_4 + 2208
3  SwiftUI                        0x847c5c OUTLINED_FUNCTION_0 + 4544
4  SwiftUI                        0x28a98 OUTLINED_FUNCTION_63 + 2384
5  SwiftUI                        0x554f8 __swift_allocate_boxed_opaque_existential_0 + 1324
6  SwiftUI                        0x2e0cc OUTLINED_FUNCTION_176 + 1220
7  SwiftUI                        0x552ec __swift_allocate_boxed_opaque_existential_0 + 800
8  SwiftUI                        0x2e0cc OUTLINED_FUNCTION_176 + 1220
9  SwiftUI                        0x552ec __swift_allocate_boxed_opaque_existential_0 + 800
10 SwiftUI                        0x2e0cc OUTLINED_FUNCTION_176 + 1220
11 SwiftUI                        0x552ec __swift_allocate_boxed_opaque_existential_0 + 800
12 SwiftUI                        0x2e0cc OUTLINED_FUNCTION_176 + 1220
13 SwiftUI                        0x552ec __swift_allocate_boxed_opaque_existential_0 + 800
14 SwiftUI                        0x2e0cc OUTLINED_FUNCTION_176 + 1220
15 SwiftUI                        0x552ec __swift_allocate_boxed_opaque_existential_0 + 800
16 SwiftUI                        0x2e0cc OUTLINED_FUNCTION_176 + 1220
17 SwiftUI                        0xd69938 OUTLINED_FUNCTION_4 + 1496
18 SwiftUI                        0xd69754 OUTLINED_FUNCTION_4 + 1012
19 SwiftUI                        0xe923dc OUTLINED_FUNCTION_35 + 140
20 SwiftUI                        0x6eaff0 OUTLINED_FUNCTION_4 + 2424
21 SwiftUI                        0x6ed92c block_destroy_helper + 44
22 SwiftUI                        0xaa378 OUTLINED_FUNCTION_326 + 6188
23 AttributeGraph                 0xb1c4 AGGraphWithUpdate + 76
24 SwiftUI                        0x4fec8 wait_for_lock + 108
25 SwiftUI                        0x1143c _MovableLockLock + 64
26 SwiftUI                        0xce2a8 sync_main_callback + 20
27 libdispatch.dylib              0x64780 _dispatch_client_callout + 16
28 libdispatch.dylib              0x10bdc _dispatch_main_queue_drain + 928
29 libdispatch.dylib              0x1082c _dispatch_main_queue_callback_4CF$VARIANT$mp + 36
30 CoreFoundation                 0x91a2c __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 12
31 CoreFoundation                 0x756c8 __CFRunLoopRun + 2080
32 CoreFoundation                 0x79da0 CFRunLoopRunSpecific + 584
33 GraphicsServices               0x1998 GSEventRunModal + 160
34 UIKitCore                      0x37180c -[UIApplication _run] + 868
35 UIKitCore                      0x371484 UIApplicationMain + 312
36 myapp                         0x1c2f5c main + 30 (AppDelegate.swift:30)
37 ???                            0x1b2184344 (Missing)

com.apple.uikit.eventfetch-thread
0  libsystem_kernel.dylib         0x1030 mach_msg2_trap + 8
1  libsystem_kernel.dylib         0x12b18 mach_msg2_internal + 76
2  libsystem_kernel.dylib         0x12db8 mach_msg_overwrite + 484
3  libsystem_kernel.dylib         0x1524 mach_msg + 20
4  CoreFoundation                 0x741c8 __CFRunLoopServiceMachPort + 156
5  CoreFoundation                 0x75360 __CFRunLoopRun + 1208
6  CoreFoundation                 0x79da0 CFRunLoopRunSpecific + 584
7  Foundation                     0x3e138 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 208
8  Foundation                     0x3e034 -[NSRunLoop(NSRunLoop) runUntilDate:] + 60
9  UIKitCore                      0x493cd8 -[UIEventFetcher threadMain] + 404
10 Foundation                     0x55c9c __NSThread__start__ + 704
11 libsystem_pthread.dylib        0x30ec _pthread_start + 116
12 libsystem_pthread.dylib        0x172c thread_start + 8

com.google.firebase.crashlytics.MachExceptionServer
0  myapp                         0x8b0314 FIRCLSProcessRecordAllThreads + 393 (FIRCLSProcess.c:393)
1  myapp                         0x8b06f4 FIRCLSProcessRecordAllThreads + 424 (FIRCLSProcess.c:424)
2  myapp                         0x8be1c4 FIRCLSHandler + 34 (FIRCLSHandler.m:34)
3  myapp                         0x8be9c8 FIRCLSMachExceptionServer + 521 (FIRCLSMachException.c:521)
4  libsystem_pthread.dylib        0x30ec _pthread_start + 116
5  libsystem_pthread.dylib        0x172c thread_start + 8

com.apple.NSURLConnectionLoader
0  libsystem_kernel.dylib         0x1030 mach_msg2_trap + 8
1  libsystem_kernel.dylib         0x12b18 mach_msg2_internal + 76
2  libsystem_kernel.dylib         0x12db8 mach_msg_overwrite + 484
3  libsystem_kernel.dylib         0x1524 mach_msg + 20
4  CoreFoundation                 0x741c8 __CFRunLoopServiceMachPort + 156
5  CoreFoundation                 0x75360 __CFRunLoopRun + 1208
6  CoreFoundation                 0x79da0 CFRunLoopRunSpecific + 584
7  CFNetwork                      0x228520 _CFURLStorageSessionDisableCache + 50860
8  Foundation                     0x55c9c __NSThread__start__ + 704
9  libsystem_pthread.dylib        0x30ec _pthread_start + 116
10 libsystem_pthread.dylib        0x172c thread_start + 8

Thread
0  libsystem_kernel.dylib         0x14f0 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1c1c _pthread_wqthread + 360
2  libsystem_pthread.dylib        0x1720 start_wqthread + 8

Thread
0  libsystem_kernel.dylib         0x14f0 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1c1c _pthread_wqthread + 360
2  libsystem_pthread.dylib        0x1720 start_wqthread + 8

Thread
0  libsystem_kernel.dylib         0x14f0 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1c1c _pthread_wqthread + 360
2  libsystem_pthread.dylib        0x1720 start_wqthread + 8

Thread
0  libsystem_pthread.dylib        0x1718 start_wqthread + 270

Crashed: OneSignal.OSOperationRepo
0  libobjc.A.dylib                0x1814 objc_msgSend + 20
1  Foundation                     0x12284 _encodeObject + 172
2  OneSignalUser                  0x12a14 __swift_destroy_boxed_opaque_existential_0 + 6928
3  OneSignalUser                  0x12a74 __swift_destroy_boxed_opaque_existential_0 + 7024
4  Foundation                     0x12620 _encodeObject + 1096
5  OneSignalOSCore                0xe634 __swift_destroy_boxed_opaque_existential_0 + 5612
6  OneSignalOSCore                0xe748 __swift_destroy_boxed_opaque_existential_0 + 5888
7  Foundation                     0x12620 _encodeObject + 1096
8  Foundation                     0x547f4 -[NSKeyedArchiver _encodeArrayOfObjects:forKey:] + 344
9  Foundation                     0x12620 _encodeObject + 1096
10 Foundation                     0xabdfc +[NSKeyedArchiver archivedDataWithRootObject:] + 100
11 OneSignalCore                  0xd69c -[OneSignalUserDefaults saveCodeableDataForKey:withValue:] + 88
12 OneSignalOSCore                0x7db4 __swift_instantiateConcreteTypeFromMangledName + 2856
13 OneSignalOSCore                0x763c __swift_instantiateConcreteTypeFromMangledName + 944
14 libdispatch.dylib              0x637a8 _dispatch_call_block_and_release + 24
15 libdispatch.dylib              0x64780 _dispatch_client_callout + 16
16 libdispatch.dylib              0xa298 _dispatch_lane_serial_drain$VARIANT$mp + 644
17 libdispatch.dylib              0xad70 _dispatch_lane_invoke$VARIANT$mp + 396
18 libdispatch.dylib              0x14dc8 _dispatch_workloop_worker_thread + 632
19 libsystem_pthread.dylib        0x1bd0 _pthread_wqthread + 284
20 libsystem_pthread.dylib        0x1720 start_wqthread + 8

com.apple.SwiftUI.AsyncRenderer
0  libsystem_kernel.dylib         0x187c __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x99ac _pthread_cond_wait$VARIANT$mp + 1236
2  SwiftUI                        0x6141c _MovableLockSyncMain + 128
3  SwiftUI                        0x5305c OUTLINED_FUNCTION_244 + 5680
4  SwiftUI                        0x12b9c OUTLINED_FUNCTION_17 + 1168
5  SwiftUI                        0xeb40 _setThreadGeometryProxyData + 232
6  SwiftUI                        0xccc8 OUTLINED_FUNCTION_12 + 668
7  SwiftUI                        0xdeb8 _MovableLockUnlock + 616
8  QuartzCore                     0x27ba0 CA::Display::DisplayLink::dispatch_items(unsigned long long, unsigned long long, unsigned long long) + 832
9  QuartzCore                     0x3733c display_timer_callback(__CFMachPort*, void*, long, void*) + 368
10 CoreFoundation                 0x74e64 __CFMachPortPerform + 172
11 CoreFoundation                 0x90020 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE1_PERFORM_FUNCTION__ + 56
12 CoreFoundation                 0x91c34 __CFRunLoopDoSource1 + 512
13 CoreFoundation                 0x757c4 __CFRunLoopRun + 2332
14 CoreFoundation                 0x79da0 CFRunLoopRunSpecific + 584
15 Foundation                     0x3e138 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 208
16 Foundation                     0x3dfe4 -[NSRunLoop(NSRunLoop) run] + 60
17 SwiftUI                        0x8d8e8 OUTLINED_FUNCTION_581 + 596
18 SwiftUI                        0x8c028 __swift_memcpy40_8 + 1896
19 Foundation                     0x55c9c __NSThread__start__ + 704
20 libsystem_pthread.dylib        0x30ec _pthread_start + 116
21 libsystem_pthread.dylib        0x172c thread_start + 8
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by nan-li** [comment]
Hi @sebastinto thank you for reporting. We are working on the fix. Have you encountered other OneSignal-related crashes currently or before `v5.1.3`?

**Comment by sebastinto** [comment]
> Hi @sebastinto thank you for reporting. We are working on the fix. 

Good to know! 

> Have you encountered other OneSignal-related crashes currently or before `v5.1.3`?

We haven't encountered any other crashes related to OneSignal on iOS so far.

**Comment by emawby** [comment]
Fix should be in #1412

**Comment by Samin-roadsurfer** [comment]
hey any news when the fix gonna publish?

**Comment by nan-li** [comment]
Hi everyone, the fix for this crash is now released in [Release 5.1.6](https://github.com/OneSignal/OneSignal-iOS-SDK/releases/tag/5.1.6). Please update and let us know if the crashes are not resolved.
